Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp

```diff
@@ -781,6 +781,18 @@ void MockRealtimeVideoSource::endApplyingConstraints()
     m_beingConfigured = false;
 }
 
+RunLoop::Timer& MockRealtimeVideoSource::ensureEmitFrameTimer()
+{
+    if (!m_emitFrameTimer) {
+        m_emitFrameTimer = makeUnique<RunLoop::Timer>(m_runLoop.get(), [weakThis = ThreadSafeWeakPtr { *this }]() {
```

Comment:
Why do we need to delay the construction of the Timer? Why couldn't we still construct the timer in the constructor?

Reply:
I reverted this change.
tl;dr: I remembered we are not allowed to call `ref()` in constructor, and it seems we could create timer when it's needed, so I moved the creation out. It seems creating `ThreadSafeWeakPtr` only invokes `weakRef()`, so it should be fine.

Reply:
You're allowed to ref in the constructor, just call `relaxAdoptionRequirement()` if needed.